integration-tests/src/test_loop/builder.rs

```diff
@@ -53,6 +59,10 @@ pub struct TestLoopBuilder {
     drop_chunks_validated_by: Option<AccountId>,
     /// Number of latest epochs to keep before garbage collecting associated data.
     gc_num_epochs_to_keep: Option<u64>,
+    /// Custom function to change the configs before constructing each client.
+    config_modifier: Option<Box<dyn Fn(&mut ClientConfig, usize)>>,
```

Comment:
Yeah well, I agree on that idea but I'm not sure how to build one right now without introducing more mess. The problem is that the default configs are not very default to begin with. So introducing a builder only gives the illusion that the config is being predictably constructed. I think the better thing to do would be to thin down TestLoopBuilder a bit so that it is more customizable, and then the client config builder would be part of that; for now I don't think ClientConfigBuilder is the best solution (it may end up being unnecessary once TestLoopBuilder is better designed; like enable_state_sync would both enable the config and enable the corresponding functionality in the builder setup code).